File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.js
/**
 * @param {number[]} prices
 * @return {number}
 */
var maxProfit = function(prices) {
    let min=prices[0];
    let maxProfit=0;
    for(let i=1;i<prices.length;i++){
        if(prices[i]<min){
            min=prices[i]
        }
        maxProfit=Math.max( maxProfit,prices[i]-min)
    }
    return maxProfit
};

File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {
        int buy=prices[0];
        int profits=0;
        for(int i=1;i<prices.length;i++){
            if(buy>prices[i]){
               buy=prices[i];
               
            }
            profits=Math.max(profits,prices[i]-buy); 
        }
        return profits;
    }
}